extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoClientRecorder.java

```diff
@@ -26,12 +27,19 @@
         return new Supplier<MongoClientSupport>() {
             @Override
             public MongoClientSupport get() {
+
                 List<ConnectionPoolListener> connectionPoolListeners = new ArrayList<>(connectionPoolListenerSuppliers.size());
                 for (Supplier<ConnectionPoolListener> item : connectionPoolListenerSuppliers) {
                     connectionPoolListeners.add(item.get());
                 }
+                final CommandListener commandListener = Arc.container().instance(CommandListener.class).get();
```

Comment:
Shouldn't we support multiple ``CommandListener``s given it looks like MongoDB supports it?

Reply:
Agree

Reply:
Agree. The limitation is on my knowledge of how to retrieve the List of CDI beans of a given type. any help?

Reply:
`Set<Bean<?>> beans = Arc.container().beanManager().getBeans(CommandListener.class);` will return all beans of type CommandListener.

Then you can get the bean from Arc by type if the name is null or by name.
You can find an example [here](https://github.com/quarkusio/quarkus/blob/master/extensions/agroal/runtime/src/main/java/io/quarkus/agroal/runtime/health/DataSourceHealthCheck.java#L29) but there may be an easier way to do it if you don't rely on CDI but instead use Jandex and retrieve all implementors of `CommandListener`.